docs/sdk/v3/guides/03-web3-development-basics.md

```diff
@@ -0,0 +1,157 @@
+* [erigon](https://github.com/ledgerwatch/erigon) - A very efficient archival node implementation
+* [Nethermind](https://github.com/NethermindEth/nethermind) - An Ethereum implementation focused on stability
+
+As achieving high availability and making sure your node is synced up all the time turns out to be quite the challenge,
+there are nodes as a service (RPC) providers that you can use, especially in production environments.
+[Chainnodes](https://www.chainnodes.org/) is a robust RPC provider with generous free tier that you can use in both
+development and production environments.
+
+### JSON-RPC Standard
+
+RPCs communicate over the [JSON-RPC](https://www.jsonrpc.org/) standard. To send requests, you take the RPC URL and
+make a POST request with a JSON body. See the below example:
+
+`POST https://mainnet.chainnodes.org/API_KEY`
+
+Body:
+
+```json
+{
+    "jsonrpc": "2.0",
+    "method": "eth_blockNumber",
+    "params": [],
+    "id": 1
+}
+```
+
+This request would respond with the following:
+
+```json
+
+```
+
+You can find the above examples, including more, in [this Postman collection](https://www.postman.com/chainnodes/workspace/uniswap-examples).
+
+To check out all possible RPC requests, head over to the [Chainnodes Docs](https://www.chainnodes.org/docs).
+
+### Client implementations
+
+As communicating over HTTP with POST requests directly can be quite difficult, especially if you want to properly
+encode and decode responses and handle failures and exponential backoffs, there are client implementations that
+can do the heavy lifting for you. They interact as SDKs that have developer friendly APIs and internally handle
+creating the proper RPC requests, sending them to the endpoint you choose and decoding the response for you.
+
+Some of the major implementations are listed below:
+
+* [ethers.js](https://github.com/ethers-io/ethers.js) - Javascript/Typescript SDK for NodeJS and the Browser. Used throughout the Uniswap Docs.
+* [wagmi](https://github.com/wagmi-dev/wagmi) and [viem](https://github.com/wagmi-dev/viem) - Javascript/Typescript, great duo for modern Web3 development in the Browser.
+* [Web3.swift](https://github.com/Boilertalk/Web3.swift) - Swift SDK for iOS apps and Backends.
+* [KEthereum](https://github.com/komputing/KEthereum) - Kotlin SDK for Android development.
+* [ethers-rs](https://github.com/gakonst/ethers-rs) - Rust SDK.
+* [ethclient](https://github.com/ethereum/go-ethereum/tree/master/ethclient) - Go SDK, part of geth, the reference Ethereum node implementation.
+
+As you can see there are lots of SDKs to make it easier to communicate via RPC with the blockchain.
+
+### Local Development
+
+To simulate RPCs and transactions locally, you can check [this guide](./02-local-development).
+
+## Indexers
+
+As RPCs are only a slim abstraction of the database that stores the blockchain data, there are certain things that you cannot
```

Comment:
This sentence is confusing. What is the "database that stores the blockchain data"?
I suggest rewording here.